glib/src/subclass/signal.rs

```diff
@@ -68,9 +68,129 @@ impl fmt::Debug for SignalInvocationHint {
     }
 }
 
+/// In-depth information of a specific signal
+#[repr(transparent)]
+pub struct SignalQuery(gobject_ffi::GSignalQuery);
+
+impl SignalQuery {
+    /// The name of the signal.
+    pub fn signal_name<'a>(&self) -> &'a str {
+        unsafe {
+            let ptr = self.0.signal_name;
+            std::ffi::CStr::from_ptr(ptr).to_str().unwrap()
+        }
+    }
+
+    /// The ID of the signal.
+    pub fn signal_id(&self) -> SignalId {
+        unsafe { SignalId::from_glib(self.0.signal_id) }
+    }
+
+    /// The interface/instance this signal can be emitted for.
```

Comment:
```suggestion
    /// The interface/class this signal can be emitted for.
```

Maybe? Or "instance type"?